docs/api/specs/webConnectionProtocol.md

```diff
@@ -0,0 +1,384 @@
+
+1. Locate a Desktop Agent interface
+2. Validate app & instance identity
+3. Persist connection details to SessionStorage
+4. Return DesktopAgent interface
+5. Disconnect
+
+### Step 1: Locate a Desktop Agent interface
+
+#### 1.1 Check for a possible navigation or refresh event
+
+Check the SessionStorage key `FDC3-Desktop-Agent-Details` for a `DesktopAgentDetails` record. If it exists, then a navigation or refresh event may have occurred and the stored data MUST be sent when attempting to establish a connection to the DA. This ensures that the window can maintain a consistent `instanceId` between navigation or refresh events within an app. If it doesn't exist then proceed to step (2).
+
+Any data stored under the `FDC3-Desktop-Agent-Details` must conform to the [DesktopAgentDetails](../ref/GetAgent#persisted-connection-data) type.
+
+Existing `DesktopAgentDetails` records MUST be used to limit discovery actions (in the next step) to the same mechanism as previously used or to skip the discovery step entirely if a `agentUrl` exists, indicating that the connection should be establish by loading the URL into a hidden iframe and initiating communication with that instead.
+
+If use of the persisted data fails to establish a connection to the DA then `getAgent()` should reject its promise with `AgentNotFound` error from the [`AgentError`](../ref/Errors#agenterror) enumeration.
+
+#### 1.2 Desktop Agent Discovery
+
+Next, attempt to discover whether Desktop Agent Preload or Desktop Agent Proxy interfaces are available, within a specified timeout. The optional `params.timeout` argument to `getAgent()` allows an application to specify the timeout that should be used, with a default value of 750ms. Discovery of Desktop Agent Preload or Desktop Agent Proxy interfaces should be conducted in parallel where possible and the timeout cancelled as soon as an interface is discovered. If a `DesktopAgentDetails` record was found in the previous step, limit discovery to the interface specified, or, if an `agentUrl` property was specified in the `DesktopAgentDetails` record, skip the discovery step entirely and proceed to the next step.
+
+To discover a Desktop Agent Preload interface, check for the presence of an `fdc3` object in the global scope (i.e. `window.fdc3`). If it exists return it immediately. If it does not exist then add a listener for the global `fdc3Ready` event with the the specified timeout, e.g.:
+
+```ts
+const discoverPreloadDA = async (timeout): Promise<DesktopAgent> => {
+  return new Promise((resolve, reject) => {
+    // if the global is already available resolve immediately
+    if (window.fdc3) {
+      resolve(window.fdc3);
+    } else {
+      //`fdc3Ready` event listener function
+      const listener = () => {
+        clearTimeout(timeout);
+        if (window.fdc3) {
+          clearTimeout(timeout);
```

Comment:
Isn't this `clearTimeout` call redundant, because it has already been called before the `if` block?

Reply:
it is, also shuffled statement order some here in case you want to take another look
Fixed in b2d2ef0bb3269a92fb62396f28a9007c14fd7b74